server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/BaseODCFlowTaskDelegate.java

```diff
@@ -310,4 +356,68 @@ public boolean cancel(boolean mayInterruptIfRunning) {
 
     protected abstract boolean cancel(boolean mayInterruptIfRunning, Long taskId, TaskService taskService);
 
+    private void setDownloadLogUrl(@NonNull Long flowInstanceId) {
+        TaskEntity taskEntity = taskService.detail(taskId);
+        TaskType taskType = taskEntity.getTaskType();
+        if (SUPPORT_DOWNLOAD_LOG_TASK_TYPES.contains(taskType)) {
+            String downloadUrl = String.format("/api/v2/flow/flowInstances/%s/tasks/log/download", flowInstanceId);
+            if (Objects.nonNull(cloudObjectStorageService) && cloudObjectStorageService.supported()) {
+                File logFile =
+                        taskService.getLogFile(taskEntity.getCreatorId(), taskId + "", taskType, OdcTaskLogLevel.ALL);
+                String fileName = TaskLogFilenameGenerator.generate(flowInstanceId);
+                try {
+                    String objectName = cloudObjectStorageService.uploadTemp(fileName, logFile);
+                    downloadUrl = cloudObjectStorageService.getBucketName() + "/" + objectName;
+                    log.info("Upload task log file to OSS, taskId={}, logFileName={}", taskId, fileName);
+                } catch (Exception e) {
+                    log.warn("Failed to upload task log file to OSS, taskId={}", taskId, e);
+                    throw new RuntimeException(
+                            String.format("Failed to upload task log file to OSS, file name: %s", fileName),
+                            e.getCause());
+                }
+            }
+            String resultJson = taskEntity.getResultJson();
+            AbstractFlowTaskResult result;
+            switch (taskType) {
```

Comment:
we may use flowTaskInstance#getResult instead of this~

Reply:
Yes, done!